mbf_costmap_nav/src/mbf_costmap_nav/free_pose_search.cpp

```diff
@@ -212,64 +223,71 @@ SearchSolution FreePoseSearch::findValidOrientation(const costmap_2d::Costmap2D&
 
 SearchSolution FreePoseSearch::search() const
 {
+  // restart markers
+  if (viz_)
+  {
+    viz_->deleteMarkers();
+  }
+
   const auto costmap2d = costmap_.getCostmap();
 
   // lock costmap so content doesn't change while adding cell costs
   boost::unique_lock<costmap_2d::Costmap2D::mutex_t> lock(*(costmap2d->getMutex()));
 
   std::unordered_set<int> in_queue_or_visited;
+  std::priority_queue<Cell, std::vector<Cell>, decltype(compare_strategy_)> queue(compare_strategy_);
 
   const std::vector<geometry_msgs::Point> footprint =
       safetyPadding(costmap_, config_.use_padded_fp, config_.safety_dist);
 
   SearchSolution sol;
+  Cell test_cell;
   sol.pose.theta = config_.goal.theta;
   bool outside_or_unknown{ false };
 
-  // initializing queue with the goal cell
-  Cell start;
-  if (costmap2d->worldToMap(config_.goal.x, config_.goal.y, start.x, start.y))
+  // enforce bounds if goal is outside the map
+  int test_cell_x, test_cell_y;
+  costmap2d->worldToMapEnforceBounds(config_.goal.x, config_.goal.y, test_cell_x, test_cell_y);
+  test_cell.x = static_cast<unsigned int>(test_cell_x);
+  test_cell.y = static_cast<unsigned int>(test_cell_y);
```

Comment:
```suggestion
  costmap2d->worldToMapEnforceBounds(config_.goal.x, config_.goal.y, test_cell.x, test_cell.y);
```
S: why create additional vars for this? calling `worldToMapEnforceBounds` should work even with uint no?

Reply:
a reference of type `int &` (not const-qualified) cannot be initialized with a value of type `unsigned int`